File: pset6/readability.py
from cs50 import get_string

text = get_string("Text: ")

def main():
    w = words()
    index = 0.0588 * (100.0 * letters() / w) - 0.296 * (100.0 * sentences() / w) - 15.8
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")
        
def letters():
    letter = 0
    for char in text:
        if char.isalpha():
            letter += 1
    return letter
    
def words():
    word = 1
    for char in text:
        if char.isspace():
            word += 1
    return word
    
def sentences():
    sentence = 0
    for char in text:
        if char == "." or char == "?" or char == "!":
            sentence += 1
    return sentence
    
if __name__ == "__main__":
    main()

File: pset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);

int main(void)
{
    string text = get_string("Text: "); // Prompt user to enter text
    int l = count_letters(text), s = count_sentences(text), w = count_words(text);
    float index = 0.0588 * (100.0*l / w) - 0.296 * (100.0*s / w) - 15.8; // Computinng rounded value of reading index
    printf("%i letter(s)\n" , l);
    printf("%i word(s)\n" , w);
    printf("%i sentences(s)\n" , s);
    if(index < 1)
        printf("Before Grade 1\n");
    else if(index > 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n" , (int)round(index)); // Prints out truncated rounded value of index
}

int count_letters(string s)
{
    int count = 0; // Stores number of letters
    for(int i = 0, n = strlen(s); i < n; i++)
        if(isalpha(s[i]))
            count++;
    return count;
}

int count_words(string s)
{
    int count = 1; // Stores number of words, by keeping track of number of spaces
    for(int i = 0, n = strlen(s); i < n; i++)
        if(isspace(s[i])) // n spaces = n + 1 words
            count++;
    return count;
}

int count_sentences(string s)
{
    int count = 0; // Stores number of sentences by keeping track of punctuation
    for(int i = 0, n = strlen(s); i < n; i++)
        switch(s[i])
        {
            case '.':
            case '?':
            case '!':
                count++;
        }
    return count;
}
